src/getg2ir.f

```diff
@@ -1,74 +1,70 @@
+C>    - BYTE 005 - 008 BYTES TO SKIP IN DATA FILE BEFORE GRIB MESSAGE
+C>    - BYTE 009 - 012 BYTES TO SKIP IN MESSAGE BEFORE LUS (LOCAL USE)
+C>    SET = 0, IF NO LOCAL USE SECTION IN GRIB2 MESSAGE.
+C>    - BYTE 013 - 016 BYTES TO SKIP IN MESSAGE BEFORE GDS
+C>    - BYTE 017 - 020 BYTES TO SKIP IN MESSAGE BEFORE PDS
+C>    - BYTE 021 - 024 BYTES TO SKIP IN MESSAGE BEFORE DRS
+C>    - BYTE 025 - 028 BYTES TO SKIP IN MESSAGE BEFORE BMS
+C>    - BYTE 029 - 032 BYTES TO SKIP IN MESSAGE BEFORE DATA SECTION
+C>    - BYTE 033 - 040 BYTES TOTAL IN THE MESSAGE
+C>    - BYTE 041 - 041 GRIB VERSION NUMBER (CURRENTLY 2)
+C>    - BYTE 042 - 042 MESSAGE DISCIPLINE
+C>    - BYTE 043 - 044 FIELD NUMBER WITHIN GRIB2 MESSAGE
+C>    - BYTE 045 -  II IDENTIFICATION SECTION (IDS)
+C>    - BYTE II+1-  JJ GRID DEFINITION SECTION (GDS)
+C>    - BYTE JJ+1-  KK PRODUCT DEFINITION SECTION (PDS)
+C>    - BYTE KK+1-  LL THE DATA REPRESENTATION SECTION (DRS)
+C>    - BYTE LL+1-LL+6 FIRST 6 BYTES OF THE BIT MAP SECTION (BMS)
+C>
+C>    PROGRAM HISTORY LOG:
+C>    - 1995-10-31 Mark Iredell
+C>    - 1996-10-31 Mark Iredell AUGMENTED OPTIONAL DEFINITIONS TO BYTE 320
+C>    - 2002-01-02 Stephen Gilbert MODIFIED FROM GETGIR TO CREATE GRIB2 INDEXES
+C>
+C>    @param[in] LUGB INTEGER UNIT OF THE UNBLOCKED GRIB FILE
+C>    @param[in] MSK1 INTEGER NUMBER OF BYTES TO SEARCH FOR FIRST MESSAGE
+C>    @param[in] MSK2 INTEGER NUMBER OF BYTES TO SEARCH FOR OTHER MESSAGES
+C>    @param[in] MNUM INTEGER NUMBER OF GRIB MESSAGES TO SKIP (USUALLY 0)
+C>    OUTPUT ARGUMENTS:
+C>    @param[out] CBUF CHARACTER*1 POINTER TO A BUFFER THAT CONTAINS INDEX
+C>    RECORDS. USERS SHOULD FREE MEMORY THAT CBUF POINTS TO, USING
+C>    DEALLOCATE(CBUF) WHEN CBUF IS NO LONGER NEEDED.
+C>    @param[out] NLEN INTEGER TOTAL LENGTH OF INDEX RECORD BUFFER IN BYTES
+C>    @param[out] NNUM INTEGER NUMBER OF INDEX RECORDS, =0 IF NO GRIB
+C>    MESSAGES ARE FOUND)
+C>    @param[out] NMESS LAST GRIB MESSAGE IN FILE SUCCESSFULLY PROCESSED
+C>    @param[out] IRET INTEGER RETURN CODE
+C>    - 0 ALL OK
+C>    - 1 NOT ENOUGH MEMORY AVAILABLE TO HOLD FULL INDEX BUFFER
+C>    - 2 NOT ENOUGH MEMORY TO ALLOCATE INITIAL INDEX BUFFER
+C>
+C>    SUBPROGRAMS CALLED:
+C>    - SKGB  SEEK NEXT GRIB MESSAGE
+C>    - IXGB2 MAKE INDEX RECORD
 C
-C SUBPROGRAMS CALLED:
-C   SKGB           SEEK NEXT GRIB MESSAGE
-C   IXGB2          MAKE INDEX RECORD
-C
-C REMARKS: SUBPROGRAM CAN BE CALLED FROM A MULTIPROCESSING ENVIRONMENT.
-C   DO NOT ENGAGE THE SAME LOGICAL UNIT FROM MORE THAN ONE PROCESSOR.
-C
-C ATTRIBUTES:
-C   LANGUAGE: FORTRAN 90
-C
-C$$$
-      USE RE_ALLOC          ! NEEDED FOR SUBROUTINE REALLOC
+C>    @note SUBPROGRAM CAN BE CALLED FROM A MULTIPROCESSING ENVIRONMENT.
+C>    DO NOT ENGAGE THE SAME LOGICAL UNIT FROM MORE THAN ONE PROCESSOR.
+C>
+C>    @author Mark Iredell @date 1995-10-31
+C>
+
+      SUBROUTINE GETG2IR(LUGB,MSK1,MSK2,MNUM,CBUF,NLEN,NNUM,NMESS,IRET)
+
+      USE RE_ALLOC          !< NEEDED FOR SUBROUTINE REALLOC
```

Comment:
Since this is not a variable, I am not sure you need to add doxygen here (or on line 67).

Reply:
I will change all, and change it to lower case.